client/post-editor/editor-page-order/index.jsx

```diff
@@ -1,75 +1,78 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
-import PureRenderMixin from 'react-pure-render/mixin';
-import isNaN from 'lodash/isNaN';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import TextInput from 'components/forms/form-text-input';
-import postActions from 'lib/posts/actions';
 import { recordEvent, recordStat } from 'lib/posts/stats';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import { editPost } from 'state/posts/actions';
 
-export default React.createClass( {
-	displayName: 'EditorPageOrder',
+class EditorPageOrder extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		postId: PropTypes.number,
+		menuOrder: PropTypes.number
+	};
 
-	mixins: [ PureRenderMixin ],
+	static defaultProps = {
+		menuOrder: 0
+	};
 
-	propTypes: {
-		menuOrder: PropTypes.oneOfType( [
-			PropTypes.number,
-			PropTypes.string
-		] )
-	},
-
-	getDefaultProps() {
-		return {
-			menuOrder: 0
-		};
-	},
-
-	updateMenuOrder( event ) {
-		let newOrder = parseInt( event.target.value );
-
-		if ( isNaN( newOrder ) ) {
-			newOrder = '';
-		}
-
-		this.editMenuOrder( newOrder );
-	},
+	constructor() {
+		super( ...arguments );
 
-	validateMenuOrder( event ) {
-		let newOrder = parseInt( event.target.value );
+		this.editMenuOrder = this.editMenuOrder.bind( this );
+	}
 
-		if ( isNaN( newOrder ) ) {
-			newOrder = 0;
+	editMenuOrder( event ) {
+		const newOrder = parseInt( event.target.value ) || 0;
```

Comment:
my favorite variable name.  each time i read it, i must listen to https://www.youtube.com/watch?v=9GMjH1nR0ds